packages/noya-designsystem/src/components/InputField.tsx

```diff
@@ -1,59 +1,108 @@
+import { CaretDownIcon } from '@radix-ui/react-icons';
 import {
+  Children,
   createContext,
+  isValidElement,
   memo,
   ReactNode,
   useCallback,
   useContext,
   useMemo,
 } from 'react';
+import { Property } from 'csstype';
 import styled from 'styled-components';
 import handleNudge from '../utils/handleNudge';
 import TextInput, { TextInputProps } from './internal/TextInput';
+import { DropdownMenu as NoyaDropdownMenu } from 'noya-designsystem';
+import Button from './Button';
 
 type LabelPosition = 'start' | 'end';
 
 const InputFieldContext = createContext<{
   labelPosition: LabelPosition;
   labelSize: number;
+  hasDropdown: boolean;
 }>({
   labelPosition: 'end',
   labelSize: 6,
+  hasDropdown: false,
 });
 
 /* ----------------------------------------------------------------------------
  * Label
  * ------------------------------------------------------------------------- */
 
-const LabelContainer = styled.label<{ labelPosition: LabelPosition }>(
-  ({ theme, labelPosition }) => ({
-    color: theme.colors.textMuted,
-    position: 'absolute',
-    top: 0,
-    right: 0,
-    bottom: 0,
-    left: 0,
-    display: 'flex',
-    alignItems: 'center',
-    pointerEvents: 'none',
-    fontWeight: 'bold',
-    fontSize: '60%',
-    opacity: 0.5,
-    userSelect: 'none',
-    ...(labelPosition === 'start'
-      ? { justifyContent: 'flex-start', paddingLeft: '6px' }
-      : { justifyContent: 'flex-end', paddingRight: '6px' }),
-  }),
-);
+const LabelContainer = styled.label<{
+  labelPosition: LabelPosition;
+  hasDropdown: boolean;
+}>(({ theme, labelPosition, hasDropdown }) => ({
+  color: theme.colors.textMuted,
+  position: 'absolute',
+  top: 0,
+  right: 0,
+  bottom: 0,
+  left: 0,
+  display: 'flex',
+  alignItems: 'center',
+  pointerEvents: 'none',
+  fontWeight: 'bold',
+  fontSize: '60%',
+  opacity: 0.5,
+  userSelect: 'none',
+  ...(labelPosition === 'start'
+    ? { justifyContent: 'flex-start', paddingLeft: '6px' }
+    : {
+        justifyContent: 'flex-end',
+        paddingRight: hasDropdown ? '16px' : '6px',
+      }),
+}));
 
 interface InputFieldLabelProps {
   children?: ReactNode;
 }
 
 function InputFieldLabel({ children = false }: InputFieldLabelProps) {
-  const { labelPosition } = useContext(InputFieldContext);
+  const { labelPosition, hasDropdown } = useContext(InputFieldContext);
+
+  return (
+    <LabelContainer labelPosition={labelPosition} hasDropdown={hasDropdown}>
+      {children}
+    </LabelContainer>
+  );
+}
+
+/* ----------------------------------------------------------------------------
+ * Dropdown
+ * ------------------------------------------------------------------------- */
+
+const DropdownContainer = styled.span(({ theme }) => ({
+  position: 'absolute',
+  right: 0,
+}));
+
+interface InputFieldDropdownProps {
```

Comment:
Here we should try to reuse the generic `items` and `onSelect` props we use in `DropdownMenu` and `ContextMenu`, rather than creating a new pattern

https://github.com/noya-app/noya/blob/5380076b51cc8e225b854cc9fecbd7f2ae751450/packages/noya-designsystem/src/components/DropdownMenu.tsx#L38-L39

Reply:
I was using numbers before so it didn't work with it but I changed it a bit so I send strings instead 👍